examples/public/misc/edit-sql-cartocss-gmaps.html

```diff
@@ -0,0 +1,109 @@
+      #container { display: flex; width: 100%; height: 100%; }
+      #map { flex: 1; margin: 10px; }
+      #controls { position: relative; width: 300px; margin: 20px; }
+      h3 { margin-bottom: 20px; }
+      button { background: #162945; width: 100%; padding: 10px; margin: 20px 0; color: #FFF; border: none; font-weight: bold; text-transform: uppercase; }
+      textarea { height: 100px; width: 100%; resize: none; }
+    </style>
+  </head>
+  <body>
+    <div id="container">
+      <div id="map"></div>
+      <div id="controls">
+        <h3>SQL</h3>
+        <textarea id="sql" rows="3">
+SELECT *
+  FROM ne_10m_populated_places_simple
+</textarea>
+        <button onclick="updateSource()"> Update Layer source</button>
+        <h3>CartoCSS</h3>
+        <textarea id="cartocss" rows="3">
+#layer {
+  marker-fill: red;
+}
+</textarea>
+        <button onclick="updateStyle()"> Update Layer style</button>
+        <p> This example only uses one style and one layer whose content is changed. </p>
+        <p>
+          Since the style and source objects are the same the wont trigger events, you need to use the promise to handle errors.
+        </p>
+        <p class="popup"></p>
+      </div>
+    </div>
+    <script>
+      const map = new google.maps.Map(document.getElementById('map'), {
+        center: {lat: 30, lng: 0},
+        zoom: 3
+      });
+      // Hide the map labels and geometry strokes
+      map.set('styles', [{
+        elementType: 'labels',
+        stylers: [{ visibility: 'off' }]
+      }, {
+        elementType: 'geometry.stroke',
+        stylers: [{ visibility: 'off' }]
+      }]);
+
+      const client = new carto.Client({
+        apiKey: 'YOUR_API_KEY',
+        username: 'cartojs-test'
+      });
+
+      const populatedPlacesSource = new carto.source.SQL(getValue('#sql'));
+      const populatedPlacesStyle = new carto.style.CartoCSS(getValue('#cartocss'));
+      const populatedPlacesLayer = new carto.layer.Layer(populatedPlacesSource, populatedPlacesStyle);
+
+      client.addLayer(populatedPlacesLayer);
+      map.overlayMapTypes.push(client.getGoogleMapsMapType(map));
+
+      function updateSource() {
+        reset('#sql');
+        populatedPlacesSource.setQuery(getValue('#sql'))
+          .catch(cartoError => {
+            error('#sql', cartoError);
+          });
+      }
+
+      function updateStyle() {
+        reset('#cartocss')
+        populatedPlacesStyle.setContent(getValue('#cartocss'))
+          .catch(cartoError => {
+            error('#cartocss', cartoError);
+          });
+      }
+
+      function getValue(id) {
+        return document.querySelector(id).value
+      }
+
+      function reset(id) {
+        document.querySelector(id).style.background = 'white';
+      }
+
+      function error(id, cartoError) {
+        document.querySelector(id).style.background = '#E57373';
+        setTimeout(() => {
```

Comment:
why this timeout?

Reply:
This is to enable the color to be changed before the alert. But I accept new ideas.

Reply:
🔥